lacework-agent-windows/templates/windows-configmap.yaml

```diff
@@ -16,28 +16,29 @@ metadata:
   {{- end }}
 data:
   config.json: |
-    {"autoupgrade": "{{ required "A value for autoupgrade is required" .Values.windowsAgent.agentConfig.autoUpgrade }}",
-     {{- with .Values.windowsAgent.agentConfig.fim }}
-     {{- if .enable }}
-     "fim": {
-             {{- if .fileIgnore }}"fileignore": {{ mustToJson .fileIgnore }}, {{- end }}
-             {{- if .filePath }}"filepath": {{ mustToJson .filePath }}, {{- end }}
-             "mode": "enable"
-             {{- if .noAtime }}, "noatime": "true" {{- end }}
-             {{- if .runAt }}, "runat": "{{ .runAt }}" {{- end }}
-             {{- if .maxscanfiles }}, "maxscanfiles": {{ .maxscanfiles }} {{- end }}
-     },
-     {{- else }}
-     "fim": {"mode": "disable"},
-     {{- end }}
-     {{- end }}
-  {{- if .Values.windowsAgent.agentConfig.proxyUrl }} "proxyurl" :"{{ .Values.windowsAgent.agentConfig.proxyUrl}}", {{- end }}
-  {{- if .Values.windowsAgent.agentConfig.registry }} "registry" :"{{ .Values.windowsAgent.agentConfig.registry}}", {{- end }}
-  {{- if .Values.windowsAgent.agentConfig.hids }} "hids": "{{ .Values.windowsAgent.agentConfig.hids}}", {{- end }}
-     "serverurl": "{{ .Values.windowsAgent.agentConfig.serverUrl}}",
-     "containers": {"mode": "enable"},
-     "tags": {"Env": "{{ .Values.windowsAgent.agentConfig.env }}"
+    {
+      "autoupgrade": "{{ required "A value for autoupgrade is required" .Values.windowsAgent.agentConfig.autoUpgrade }}",
+      {{- with .Values.windowsAgent.agentConfig.fim }}
+      {{- if .enable }}
+      "fim": {
+              {{- if .fileIgnore }}"fileignore": {{ mustToJson .fileIgnore }}, {{- end }}
+              {{- if .filePath }}"filepath": {{ mustToJson .filePath }}, {{- end }}
+              "mode": "enable"
+              {{- if .noAtime }}, "noatime": "true" {{- end }}
+              {{- if .runAt }}, "runat": "{{ .runAt }}" {{- end }}
+              {{- if .maxscanfiles }}, "maxscanfiles": {{ .maxscanfiles }} {{- end }}
+      },
+      {{- else }}
+      "fim": {"mode": "disable"},
+      {{- end }}
+      {{- end }}
+      {{- if .Values.windowsAgent.agentConfig.proxyUrl }} "proxyurl" :"{{ .Values.windowsAgent.agentConfig.proxyUrl}}", {{- end }}
+      {{- if .Values.windowsAgent.agentConfig.registry }} "registry" :"{{ .Values.windowsAgent.agentConfig.registry}}", {{- end }}
+      {{- if .Values.windowsAgent.agentConfig.hids }} "hids": "{{ .Values.windowsAgent.agentConfig.hids}}", {{- end }}
+      "serverurl": "{{ .Values.windowsAgent.agentConfig.serverUrl}}",
+      "containers": {"mode": "enable"},
+      "tags": {"Env": "{{ .Values.windowsAgent.agentConfig.env }}"
               {{- if .Values.windowsAgent.agentConfig.kubernetesCluster }}, "KubernetesCluster" :"{{ .Values.windowsAgent.agentConfig.kubernetesCluster}}" {{- end }}
-             }
+              }
```

Comment:
The whitespace within the Helm template is aligned but the formatting directives do not produce well formatted output. For example, running `helm template --set windowsAgent.agentConfig.accessToken=foo -s templates/windows-configmap.yaml .` gives the following configmap:

```
# Source: lacework-agent-windows/templates/windows-configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: lacework-windows-config
  namespace: default
  labels:
    tier: monitoring
    app: lacework-agent-windows
data:
  config.json: |
    {
      "autoupgrade": "enable",
      "fim": {
              "mode": "enable"
      }, "registry" :"enable", "hids": "enable",
      "serverurl": "https://api.lacework.net",
      "containers": {"mode": "enable"},
      "tags": {"Env": ""
              }
    }
```

There are two options for fixing this. Either move all `if/end` statements to being 3 lines, such as:

```
{{- if .Values.windowsAgent.agentConfig.registry }}
      "registry" :"{{ .Values.windowsAgent.agentConfig.registry}}",
{{- end }}
```

Or change the "leading whitespace removal" directives, like so:

```
      {{ if .Values.windowsAgent.agentConfig.proxyUrl }} "proxyurl" :"{{ .Values.windowsAgent.agentConfig.proxyUrl}}", {{- end }}
```

Reply:
Sorry, I am not familiar with that at first. I used the ```helm template``` to generate an output file and used  ``` yamllint ``` to check the output and it showed it is successful. 

Thank for your suggestions and really appreciated that you can spend some time to help me.
I used this command line to test:
```
helm template -s templates/windows-configmap.yaml --set windowsAgent.agentConfig.accessToken=foo --set windowsAgent.agentConfig.proxyUrl=foo --set windowsAgent.agentConfig.env=<YOUR_COSTOMIZED_ENV> --set windowsAgent.agentConfig.kubernetesCluster=<YOUR_COSTOMIZED_KUBERNETESCLUSTER> lacework-agent-windows
```
# Source: lacework-agent-windows/templates/windows-configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: lacework-windows-config
  namespace: default
  labels:
    tier: monitoring
    app: lacework-agent-windows
data:
  config.json: |
    {
      "autoupgrade": "enable",
      "fim": {
              "mode": "enable"
      },
      "proxyurl" :"foo",
      "registry" :"enable",
      "hids": "enable",
      "serverurl": "https://api.lacework.net",
      "containers": {"mode": "enable"},
      "tags": {
              "Env": "<YOUR_COSTOMIZED_ENV>",
              "KubernetesCluster": "<YOUR_COSTOMIZED_KUBERNETESCLUSTER>"
      }
    }
 ```
It is good now.